internal/docstore/driver/driver.go

```diff
@@ -104,10 +103,21 @@ type Filter struct {
 	Value     interface{} // the value to compare using the operation
 }
 
-// A DocumentIterator is an iterator that can iterator through the results (for Get
-// action), and returns the next page that can be passed into a Query as a
-// starting point.
+// A DocumentIterator iterates through the results (for Get action), and returns
+// the next page that can be passed into a Query as a starting point.
 type DocumentIterator interface {
-	Next(doc Document) error
+
+	// Next get the next item in the query result and calls decode to decode into
+	// Document.
+	//
+	// When it is called when the iterator is at the end of a page, it should go
+	// fetch another page if it is possible, though it does not need to guarantee
+	// to have a result.
+	//
+	// When there is no result to return, returns a io.EOF.
+	Next(context.Context, Document) error
+
+	// Stop cleans up resources of the iterator, if any. It is only expected to be
+	// called when user wants to terminate the iterator early.
```

Comment:
How about:

```
// Stop terminates the iterator before Next return io.EOF, allowing any cleanup needed.
// After Stop is called, Next should return an error.
```

?

Reply:
I did the latter part (next returning io.EOF when stop is called) through the portable API.